includes/service-providers/campaign_monitor/class-newspack-newsletters-campaign-monitor.php

```diff
@@ -328,75 +327,64 @@ public function content( $post_id ) {
 	/**
 	 * Send a campaign.
 	 *
-	 * @param string  $new_status New status of the post.
-	 * @param string  $old_status Old status of the post.
-	 * @param WP_POST $post Post to send.
+	 * @param WP_Post $post Post to send.
 	 *
-	 * @throws Exception Error message if sending fails.
+	 * @return true|WP_Error True if the campaign was sent or error if failed.
 	 */
-	public function send( $new_status, $old_status, $post ) {
+	public function send( $post ) {
 		$post_id = $post->ID;
 
-		// Only run if the current post is a newsletter.
-		if ( ! Newspack_Newsletters::validate_newsletter_id( $post_id ) ) {
-			return;
-		}
+		$api_key   = $this->api_key();
+		$client_id = $this->client_id();
 
-		// Only run if the current service provider is Campaign Monitor.
-		if ( 'campaign_monitor' !== get_option( 'newspack_newsletters_service_provider', false ) ) {
-			return;
+		if ( ! $api_key ) {
+			return new WP_Error(
+				'newspack_newsletter_error',
+				__( 'No Campaign Monitor API key available.', 'newspack-newsletters' )
+			);
+		}
+		if ( ! $client_id ) {
+			return new WP_Error(
+				'newspack_newsletter_error',
+				__( 'No Campaign Monitor Client ID available.', 'newspack-newsletters' ) 
+			);
 		}
 
-		if ( ( 'publish' === $new_status && 'publish' !== $old_status ) || ( 'future' === $new_status && 'future' !== $old_status ) ) {
-			try {
-				$api_key   = $this->api_key();
-				$client_id = $this->client_id();
-
-				if ( ! $api_key ) {
-					throw new Exception( __( 'No Campaign Monitor API key available.', 'newspack-newsletters' ) );
-				}
-				if ( ! $client_id ) {
-					throw new Exception( __( 'No Campaign Monitor Client ID available.', 'newspack-newsletters' ) );
-				}
-
-				$cm_campaigns = new CS_REST_Campaigns( null, [ 'api_key' => $api_key ] );
-				$args         = $this->format_campaign_args( $post_id );
-				$send_date    = 'future' === $new_status ? $post->post_date : 'Immediately';
-
-				// Set the current user's email address as the email to receive sent confirmation.
-				$current_user       = wp_get_current_user();
-				$confirmation_email = $current_user->user_email;
+		$cm_campaigns = new CS_REST_Campaigns( null, [ 'api_key' => $api_key ] );
+		$args         = $this->format_campaign_args( $post_id );
+		$send_date    = 'future' === $new_status ? $post->post_date : 'Immediately';
```

Comment:
`$new_status` is not defined anymore.

Reply:
Good catch! a0936876f1dc807d77a93cc37842f396747b660a updates the method to always send immediately. We don't have to manage scheduling if the method is only called when changing from draft to publish, which can be scheduled through WordPress.